_install-and-configure/additional-plugins/ingest-attachment-plugin.md

```diff
@@ -0,0 +1,228 @@
+---
+layout: default
+title: Ingest-attachment plugin
+parent: Installing plugins
+nav_order: 20
+
+---
+
+# Ingest-attachment plugin
+
+The Ingest-attachment plugin enables OpenSearch to extract content and other information from files using the Apache text extraction library [Tika](https://tika.apache.org/).
+Supported document formats include PPT, PDF, RTF, ODF and many more ([Tika Supported Document Formats](https://tika.apache.org/2.9.2/formats.html)).
+
+The input field must be a base64-encoded binary.
+
+## Installing the plugin
+
+Install the `ingest-attachment` plugin using the following command:
+
+```sh
+./bin/opensearch-plugin install ingest-attachment
+```
+
+## Attachment processor options
+
+| Name | Required | Default | Description |
+| :--- | :--- | :--- | :--- |
+| `field` | yes | - | The field to get base64 encoded binary from. |
+| `target_field` | no | attachment | The field that holds the attachment information. |
+| `properties` | no | all properties | An array of properties, which should be stored. Can be `content`, `language`, `date`, `title`, `author`, `keywords`, `content_type`, `content_length`. |
+| `indexed_chars` | no | `100_000` | The number of character used for extraction to prevent fields from becoming to large. Use `-1` for no limit. |
+| `indexed_chars_field` | no | `null` | The field name from which you can overwrite the number of chars being used for extraction, for example, `indexed_chars`. |
+| `ignore_missing` | no | `false` | When `true`, the processor exits without modifying the document when the specified field doesn't exist. |
+
+## Example
+
+The following steps show how to get started with the Ingest-attachment plugin.
+
+### Create an index to store your attachments
+
+The following command creates an index for storing your attachments:
+
+```json
+PUT /example-attachment-index
+{
+  "mappings": {
+    "properties": {}
+  }
+}
+```
+
+### Create a pipeline with attachment processor
+
+The following command creates a pipeline which contains the attachment processor:
+
+```json
+PUT _ingest/pipeline/attachment
+{
+  "description" : "Extract attachment information",
+  "processors" : [
+    {
+      "attachment" : {
+        "field" : "data"
+      }
+    }
+  ]
+}
+```
+
+### Store an attachment
+
+Convert the attachment to base64 string, to pass it as `data`.
+In this example the Unix-like system `base64` command converts the file `lorem.rtf`:
+
+```sh
+base64 lorem.rtf
+```
+
+Alternatively you can use Node.js to read the file to `base64`, as shown in the following commands:
+
+```typescript
+import * as fs from "node:fs/promises";
+import path from "node:path";
+
+const filePath = path.join(import.meta.dirname, "lorem.rtf");
+const base64File = await fs.readFile(filePath, { encoding: "base64" });
+
+console.log(base64File);
+```
+
+The following base64 string is for an `.rtf` file containing the text
```

Comment:
Please clarify "the text". What text, specifically? Also, this line should end in a colon.